test/validation.cjs

```diff
@@ -101,6 +101,28 @@ describe('validation tests', () => {
       failCalled.should.equal(true);
     });
 
+    it("doesn't fail if implied key exists with value 0", () => {
+      let failCalled = false;
+      yargs('--foo --bar 0')
+        .implies('foo', 'bar')
+        .fail(msg => {
+          failCalled = true;
```

Comment:
You could probably remove `failCalled` declaration/update/assertion and instead use `chai.expect.fail()` in the fail callback. `expect` should already be imported from chai.

```js
.fail(() => {
  expect.fail()
}
```